code/src/layouts/header/components/header-toolbar/header-cart-button/HeaderCartButton.tsx

```diff
@@ -0,0 +1,58 @@
+import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
+
+import CartDrawer from "@/containers/cart-drawer/CartDrawer";
+
+import AppBadge from "@/components/app-badge/AppBadge";
+import AppIconButton from "@/components/app-icon-button/AppIconButton";
+import AppTooltip from "@/components/app-tooltip/AppTooltip";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { useDrawerContext } from "@/context/drawer/DrawerContext";
+import useGetCart from "@/hooks/use-get-cart/useGetCart";
+import calculateCartItemsCounter from "@/utils/calculate-cart-items-counter/calculateCartItemsCounter";
+
+const HeaderCartButton = () => {
+  const { openDrawer } = useDrawerContext();
+
+  const { data } = useGetCart();
+
+  const handleOpenCart = () => {
+    openDrawer(<CartDrawer />);
+  };
+
+  const cartItemsCount = calculateCartItemsCounter(data.items);
+
+  const badgeContent = (
+    <AppTypography variant="caption-small" data-testid="cart-items-count">
+      {cartItemsCount}
+    </AppTypography>
+  );
+
+  const cartIcon = cartItemsCount ? (
+    <AppBadge
+      badgeContent={badgeContent}
+      variant="contained"
+      size="small"
+      className="header__toolbar-cart-badge"
+      data-testid="header-cart-badge"
+    >
+      <ShoppingCartIcon className="header__toolbar-icon" fontSize="medium" />
+    </AppBadge>
+  ) : (
+    <ShoppingCartIcon className="header__toolbar-icon" fontSize="medium" />
```

Comment:
Create separate variable for `shoppingCartIcon`

Reply:
Thanks, done